include/wrenbind17/pop.hpp

```diff
@@ -255,6 +260,11 @@ namespace wrenbind17 {
             return std::string(wrenGetSlotString(vm, idx));
         }
 
+        template <> inline std::string_view getSlot(WrenVM* vm, int idx) {
+            validate<WrenType::WREN_TYPE_STRING>(vm, idx);
+            return std::string_view(wrenGetSlotString(vm, idx));
```

Comment:
This may be very dangerous. The string_view would not own the memory.

Therefore, if you have a C++ function that accepts string_view, and stores the string_view for a future use, then that may cause a segmentation fault. As far as I know Wren only guarantees that the raw bytes of that slot (string) is valid only during the call of the C++ function. The garbage collection may run after calling the function and the string_view points to an unknown memory.

This is fine the other way around. Wren makes a copy of the string/string_view you give it.

I am fine with accepting the risk, but could you update the documentation here? -> https://github.com/matusnovak/wrenbind17/blob/master/docs/content/Tutorial/types.md Maybe adding some section explaining this danger. I don't mind the exact paragraphs, I can leave that to you. Plus a brief comment on this line that yes it is a risk.